ls_pre_commit_hooks/plantuml-renderer.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+if [[ -n $CIRCLECI ]]; then
+ echo "Only to be executed locally, as a pre-commit hook."
+ exit 0;
+fi
+
+if ! command -v plantuml &>/dev/null; then
+  if command -v brew &>/dev/null; then
+    brew install plantuml
+  else
+    >&2 echo 'plantuml cannot be installed with brew. Get plantuml from https://plantuml.com/download.'
+  fi
+fi
+
+for puml_file in "${@}"; do
+  plantuml -tsvg "$puml_file" -o "images"
+  FILEDIR="${puml_file%/*}/images"
+  FILENAME="${puml_file##*/}"
+  SVG_FILE="$FILEDIR/${FILENAME%.*}.svg"
+  echo "" >> $SVG_FILE # add new line for tidy-xml conflict
+  git add "$SVG_FILE"
+done
```

Comment:
```suggestion

changed_files=0
for puml_file in "${@}"; do
  plantuml -tsvg "$puml_file" -o "images"
  FILEDIR="${puml_file%/*}/images"
  FILENAME="${puml_file##*/}"
  SVG_FILE="$FILEDIR/${FILENAME%.*}.svg"
  echo "" >> $SVG_FILE # add new line for tidy-xml conflict
  # Check if the svg file is dirty in the git unstaged changes
  if ! git diff -q -- "$SVG_FILE" ; then
    changed_files=$((changed_files+1))
  fi
done

# Exit with non-zero code if any of the svg files was changed
exit $changed_files
```

Reply:
(`--quiet` instead of `-q`)

Checking the diff and failing is cleaner than git add ing the file IMO.

Reply:
I think it only for update, also how can I commit the changes?

1. It generate puml
2. It detect diff -> fail
3. I want to recommit
4. It still will regenerate
5. it detect diff -> fail again
6. and repeat
I am pretty sure I need to compare the generated with the current one to make this work..? Using some kind of tmp dir.?

It also cant handle the one that newly created. Since git diff wont give anything if its new file right?

Reply:
It will usually work like this:

```
$ nano mydrawing.plantuml
$ git add mydrawing.plantuml
$ git commit -m "Editing my plantuml"
pre-commit will run here and fail!
$ git status
will show the updated svg file
$ git add mydrawing.plantuml  # after a human looks at it to confirm that it's good
$ git commit -m "Editing my plantuml"
pre-commit will now re-generate the svg but there will be no diff, so it will not fail
$ git status
clean!
$ git push
```

Reply:
@ls-jad-elkik updated with:

- Generate the svg at `_tmp_` dir
- Check existing svg file
  - If exist then:
     - Check if the file is identical
       - If it is the no changedfile added
       - If different then move and overwrite existing svg file, mark as changedfile  
  - If not exist then:
     - Move the generated svg file
     - Stage it
     - Mark as changed file

Let me know what you think about this approach.

Reply:
okay my mistake.. when I commit I used IDE, and forgot to remove the reformat.. thats why it keep saying it have difference...

Updated. Also added the `git add -N` to move the untracked to unstaged